src/sentry/search/events/fields.py

```diff
@@ -1314,6 +1314,13 @@ def with_default(default, argument):
     return argument
 
 
+class SnQLColumnOrAlias(FunctionArg):
+    def normalize(self, value, _):
+        if value is None:
+            raise InvalidFunctionArgument("a column or alias is required")
+        return value
```

Comment:
This should more closely resemble the existing `NumericColumn` and `NumericColumnNoLookup`.

I believe that we've agreed on that the `normalize` method here should just do the necessary validations without resolving the column into `SnQL` and letting that resolution happen within `resolve_function` like you have below.

Reply:
And just to keep it consistent, you can just use the same name as the classes you are trying to replicate and prefix it with SnQL. So `SnQLNumericColumnNoLookup`, that will help clean up after the migration

Reply:
I've renamed to `SnQLNumericColumnNoLookup` and updated to extend `NumericColumn`. `NumericColumnNoLookup` itself didn't have any validation, so just extending `NumericColumn` to inherit validation should be sufficient.